OpenRA.Game/Network/LocalizedMessage.cs

```diff
@@ -0,0 +1,130 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2021 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Fluent.Net;
+using OpenRA.Widgets;
+
+namespace OpenRA.Network
+{
+	[Flags]
+	public enum SupportedTypes
+	{
+		String = 0,
+		Number = 1,
+	}
+
+	public class Argument
+	{
+		public readonly string Key;
+		public readonly string Value;
+		public readonly SupportedTypes Type;
+
+		public Argument() { }
+
+		public Argument(string key, object value)
+		{
+			Key = key;
```

Comment:
This is currently fragile against unexpected value types.

Something like this would make it consistent with FTL:
```csharp
Key = key;
Value = value.ToString();
Type = value is sbyte || value is short || value is int || value is long ||
    value is byte || value is ushort || value is uint || value is ulong ||
    value is float || value is double || value is decimal ?
    SupportedTypes.Number : SupportedTypes.String;
```